js/src/main/scala/JSVyxal.scala

```diff
@@ -10,33 +9,35 @@ import scala.scalajs.js.JSConverters.*
 @JSExportTopLevel("Vyxal")
 object JSVyxal:
   @JSExport
-  def execute(code: String, inputs: String, flags: String): String =
+  def execute(
+      code: String,
+      inputs: String,
+      flags: String,
+      printFunc: String
+  ): Unit =
     // todo take flags to set settings
     // todo take functions to print to custom stdout and stderr
-    val output = document.getElementById("output")
-    // cast output object to textarea
-    val textarea = output.asInstanceOf[dom.html.TextArea]
-    textarea.value = ""
-    val settings =
-      Settings(online = true)
+    val settings = flags.foldLeft(Settings(online = true))(_.withFlag(_))
     val globals = Globals(
-      settings = settings
+      settings = settings,
+      printFn = (x: VAny) =>
+        val f = js
+          .eval(s"$printFunc")
+          .asInstanceOf[js.Function1[String, Unit]]
+        f.apply(if x.toString.isEmpty then "" else x.toString)
```

Comment:
You used Copilot here, didn't you?

Reply:
This is the one time I searched through docs to write a cursed mess :p

Reply:
Oh ok, just asking because `f.apply(if x.toString.isEmpty then "" else x.toString)` is the same as `f.apply(x)`

Reply:
The empty string check is an artifact from something else I tried lol